mlrun/api/utils/clients/iguazio.py

```diff
@@ -0,0 +1,78 @@
+import copy
+import typing
+
+import requests.adapters
+import urllib3
+
+import mlrun.api.schemas
+import mlrun.api.utils.projects.remotes.member
+import mlrun.errors
+import mlrun.utils.singleton
+from mlrun.utils import logger
+
+
+class Client(metaclass=mlrun.utils.singleton.Singleton,):
+    def __init__(self) -> None:
+        super().__init__()
+        http_adapter = requests.adapters.HTTPAdapter(
+            max_retries=urllib3.util.retry.Retry(total=3, backoff_factor=1)
+        )
+        self._session = requests.Session()
+        self._session.mount("http://", http_adapter)
+        self._api_url = mlrun.mlconf.iguazio_api_url
+
+    def get_grafana_service_url_if_exists(
```

Comment:
```suggestion
    def get_grafana_service_url(
```
Minor - I think  the type hint and possible a docstring can make it clear it returns None if there isn't any and that's expected.
If you really want the name to signify the "best-effort-ness" maybe `try_get_grafana_service_url`
Simply because it's more concise but conveys the same meaning